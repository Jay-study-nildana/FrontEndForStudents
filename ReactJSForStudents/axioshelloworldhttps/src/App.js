import './App.css';

//enguerrand-blanchy-3oADW0Ptj8c-unsplash
import GetSomething from './components/GetSomething';
import GetSomethingWithButton from './components/GetSomethingWithButton';
import PostSomethingMongo from './components/PostSomethingMongo';
import localImageDiamondMoon from "./images/enguerrand-blanchy-3oADW0Ptj8c-unsplash.jpg";
import GetSomethingMongo from "./components/GetSomethingMongo";
let string1 = `this is coming from App JS`;



function App() {
  return (
    <div className="App">
      {/* <div className="text-center hero my-5">
        <GetSomething />
      </div> */}
      <div className="text-center hero my-5">
        <GetSomethingWithButton />
      </div>   
      <div className="text-center hero my-5">
        <GetSomethingMongo />
      </div>    
      <div className="text-center hero my-5">
        <PostSomethingMongo />
      </div>             
      <div className="text-center hero my-5">
        {/* <img className="mb-3 app-logo" src={logo} alt="React logo" width="120" /> */}
        <h1 className="mb-4">{string1}</h1>   
        <hr></hr>
        <div className='container-fluid'>
          <img src={localImageDiamondMoon} className="img-fluid" alt="..."></img>  
        </div>
      </div>       
      {/* <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header> */}
    </div>
  );
}

export default App;
