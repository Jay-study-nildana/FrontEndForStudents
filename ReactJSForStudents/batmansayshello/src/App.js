import { useEffect } from "react"; //automatically trigger things on page load
import { useState } from "react"; //basic state management from react

import "./styles.css";

//here, you can substitute the endpoint to any other api of your choice
const apiurl = `https://official-joke-api.appspot.com/random_joke`;
// i just want to show some emoticons
const loadingsymbols = "🐃🐃🐒🐒🦒🦒";

export default function App() {
  //simple loading indicator
  const [simpleloader, setsimpleloader] = useState("click to load joke");
  const [joketype, setjoketype] = useState("Joke Type");
  const [jokesetup, setjokesetup] = useState("Setup");
  const [jokepunchline, setjokepunchline] = useState("Punchline");
  const [jokeid, setjokeid] = useState("Joke ID");
  const [totaljokes, settotaljokes] = useState(0);

  //function that will load the joke
  async function LoadTheJoke() {
    console.log(`joke about to load`);
    setsimpleloader("joke loading...");
    setjoketype(loadingsymbols);
    setjokepunchline(loadingsymbols);
    setjokesetup(loadingsymbols);
    setjokeid(loadingsymbols);
    const jokeapiresponseobject = await fetch(apiurl);
    const jokedata = await jokeapiresponseobject.json();
    setsimpleloader("click to load joke");
    setjoketype(jokedata.type);
    setjokepunchline(jokedata.setup);
    setjokesetup(jokedata.punchline);
    setjokeid(jokedata.id);
    settotaljokes((c) => c + 1); //lambda function
    console.log(jokedata);
  }

  //function and dependency array
  //as of now, it is empty

  useEffect(function () {
    //call the default advice thing
    //console.log(`useeffect has completed`)
    //if you want to, you could call the joke load function and populate an initial joke
    //I will leave that as an exercise for you to try.
  }, []);

  return (
    <div className="divfirst">
      <header>
        <h1 className="header-h1">Let's show a random joke</h1>
      </header>
      <main>
        <div className="section1">
          <p>{jokepunchline}</p>
          <h3>{jokesetup}</h3>
        </div>

        <div className="section1">
          <p>
            type: {joketype} id: {jokeid}
          </p>
        </div>

        <button onClick={LoadTheJoke}>{simpleloader}</button>

        <p>Total Jokes Read : {totaljokes}</p>
      </main>
      <footer>
        <p>
          code by Jay (Vijayasimha BR).{" "}
          <a href="https://stories.thechalakas.com/">website</a> and{" "}
          <a href="https://github.com/Jay-study-nildana">github</a> and{" "}
          <a href="https://www.youtube.com/channel/UCJJVulg4J7POMdX0veuacXw">
            YouTube Tutorials
          </a>
        </p>
      </footer>
    </div>
  );
}
