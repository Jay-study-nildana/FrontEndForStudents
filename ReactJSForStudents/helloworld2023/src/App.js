// import "./styles.css";

// export default function App() {
//   return (
//     <div className="App">
//       <h1>Hello CodeSandbox</h1>
//       <h2>Start editing to see some magic happen!</h2>
//     </div>
//   );
// }

import { Routes, Route } from "react-router-dom";
import Home from "./components/Home";
import Layout from "./components/Layout";
import NoMatch from "./components/NoMatch";
import PassingObjects from "./components/PassingObjects";
import InputStuff from "./components/InputStuff";
import ImageSlider from "./components/ImageSlider";

export default function App() {
  return (
    <div className="App">
      <hr></hr>
      <Routes>
        <Route path="/" element={<Layout />}>
          <Route index element={<Home />} />
          <Route path="/passingobjects" element={<PassingObjects />} />
          <Route path="/input" element={<InputStuff />} />
          <Route path="/slideshow" element={<ImageSlider />} />
          {/* <Route path="about" element={<About />} /> */}
          {/* <Route path="dashboard" element={<Dashboard />} /> */}
          {/* Using path="*"" means "match anything", so this route
                acts like a catch-all for URLs that we don't have explicit
                routes for. */}
          <Route path="*" element={<NoMatch />} />
        </Route>
      </Routes>
      {/* <FooterContact /> */}
      {/* <Ameesha /> */}
    </div>
  );
}
