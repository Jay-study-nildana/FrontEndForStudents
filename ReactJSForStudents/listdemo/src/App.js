import { Routes, Route } from "react-router-dom";
import Home from "./components/Home";
import Layout from "./components/Layout";
import ListSimple from "./components/ListSimple";
import ListSimple2 from "./components/ListSimple2";
import ListBatman1 from "./components/ListBatman1";
import ListBatman2 from "./components/ListBatman2";
import ListMovies from "./components/ListMovies";
import NoMatch from "./components/NoMatch";
import MovieDetailsPage from "./components/MovieDetailsPage";

function App() {
  return (
    <div className="App">
      <hr></hr>
      <Routes>
        <Route path="/" element={<Layout />}>
          <Route index element={<Home />} />
          <Route path="/listsimple" element={<ListSimple />} />
          <Route path="/listsimple2" element={<ListSimple2 />} />
          <Route path="/listbatman1" element={<ListBatman1 />} />
          <Route path="/listbatman2" element={<ListBatman2 />} />
          <Route path="/listmovies" element={<ListMovies />} />
          <Route path="/listmovies/:movieid" element={<MovieDetailsPage/>} />
          {/* Using path="*"" means "match anything", so this route
                acts like a catch-all for URLs that we don't have explicit
                routes for. */}
          <Route path="*" element={<NoMatch />} />
        </Route>
      </Routes>
      {/* <FooterContact /> */}
      {/* <Ameesha /> */}
    </div>
  );
}

export default App;
