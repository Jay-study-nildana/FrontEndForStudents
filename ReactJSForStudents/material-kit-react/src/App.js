/**
=========================================================
* Material Kit 2 React - v2.0.0
=========================================================

* Product Page: https://www.creative-tim.com/product/material-kit-react
* Copyright 2021 Creative Tim (https://www.creative-tim.com)

Coded by www.creative-tim.com

 =========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
*/

import { useEffect } from "react";

// react-router components
import { Routes, Route, Navigate, useLocation } from "react-router-dom";

// @mui material components
import { ThemeProvider } from "@mui/material/styles";
import CssBaseline from "@mui/material/CssBaseline";

// Material Kit 2 React themes
import theme from "assets/theme";
import Presentation from "layouts/pages/presentation";

// Material Kit 2 React routes
import routes from "routes";

export default function App() {
  const { pathname } = useLocation();

  // Setting page scroll to 0 when changing the route
  useEffect(() => {
    document.documentElement.scrollTop = 0;
    document.scrollingElement.scrollTop = 0;
  }, [pathname]);

  const getRoutes = (allRoutes) =>
    allRoutes.map((route) => {
      if (route.collapse) {
        return getRoutes(route.collapse);
      }

      if (route.route) {
        return <Route exact path={route.route} element={route.component} key={route.key} />;
      }

      return null;
    });

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <Routes>
        {getRoutes(routes)}
        <Route path="/presentation" element={<Presentation />} />
        <Route path="*" element={<Navigate to="/presentation" />} />
      </Routes>
    </ThemeProvider>
  );
}
