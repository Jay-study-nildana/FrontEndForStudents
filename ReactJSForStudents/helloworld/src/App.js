import logo from './logo.svg';
import './App.css';
import localImageDiamondMoon from "./images/vijayasimha-br-UjhiDnhHBrM-unsplash.jpg";
import ShowImageAgain from "./components/ShowImageAgain";
import ShowSomeStuff from "./components/ShowSomeStuff"

let string1 = `Showing a Local Image From Local Directory`;
let somemessage1 = "keep scrolling for more things";
let someObject1 = {
  name : `Jay`,
  profession : `Freelance Coding Tutor`,
  purpose : `To be my own boss`,
  totalExperience : 10,
  numberOfHoursSleepingEveryDay : 10,
  lifeMotto : `Work as little as possible`
}
let someObject2 = {
  name : `Batman`,
  profession : `SuperHero`,
  purpose : `fight crime`,
  totalExperience : 69,
  numberOfHoursSleepingEveryDay : 0,
  lifeMotto : `I am Batman`
}


function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React. 
        </a>
        <p>Note : Scroll down to see a simple image display.</p>
        <p>{somemessage1}</p>
      </header>
      <div className="text-center hero my-5">
        {/* <img className="mb-3 app-logo" src={logo} alt="React logo" width="120" /> */}
        <h1 className="mb-4">{string1}</h1>   
        <img src={localImageDiamondMoon} className="img-fluid" alt="..."></img>
      </div> 
      <div>
        <ShowImageAgain/>
      </div>   
      <div>
        <p>Hello there</p>
        <ShowSomeStuff someObject1 = {someObject1} someObject2 = {someObject2} />
        <p>Hello there End</p>
      </div>      
      <ShowSomeStuff someObject1 = {someObject1} someObject2 = {someObject2} />  
    </div>
  );
}

export default App;
