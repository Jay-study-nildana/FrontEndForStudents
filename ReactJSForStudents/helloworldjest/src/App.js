import logo from './logo.svg';
import './App.css';
import AnotherComponent from './AnotherComponent';

function App() {
  return (
    <div className="App">
      {/* change the following line after taking snapshots to see tests fail */}
      <button>ShowBro123dfdfdfdfdfdfdfdfdf</button>
    </div>
  );
}

export default App;
