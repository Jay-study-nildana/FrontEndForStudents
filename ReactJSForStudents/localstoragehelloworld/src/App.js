import logo from './logo.svg';
import './App.css';
import { useState } from 'react';
import { useEffect } from 'react';


const ShowSomeStuff = () => {

  const [quoteContent, setquoteContent] = useState('');
  const [quoteContentMimic, setquoteContentMimic] = useState('');

  var ls = require('local-storage');

  useEffect(() => {
    console.log(quoteContent);
    setquoteContentMimic(quoteContent);
    console.log(quoteContentMimic);

    if (quoteContent.length === 0) {
      let someMessage = `You have not yet typed anything. Type something`;
      setquoteContentMimic(someMessage);
    }

  }, [quoteContent, quoteContentMimic]);

  function clearEverything() {

    ls.clear();
    console.log(`local storage has been cleared`);

  }

  function addStuff() {
    //foo is the key
    //bar is the value
    //standard key value things that you see in dictionaries
    ls('foo', 'bar');
    ls.set('batman', 'Bruce Wayne');
    ls.set('inputentered',quoteContentMimic);
    // <- 'bar'
    console.log(`added stuff`);
  }

  function showStuff() {
    console.log(ls.get('batman'))
    console.log(ls('foo'));
    console.log(ls.get('inputentered'))
  }

  const stuff = (
    <div className="text-center">
      <p>Here, we are just showing some things.</p>
      <hr></hr>
      <div className="input-group mb-3">
        <div>
          <input
            type="text"
            className="form-control"
            placeholder="enter anything you want"
            onChange={e => setquoteContent(e.target.value)}
            value={quoteContent}
          />
          <p>You are typing : {quoteContentMimic}</p>
        </div>
        <hr></hr>
        <br></br>
        <div>
          {/* <Button onClick={clearEverything}>Clear Everything</Button> */}

        </div>
      </div>
      <div className="text-center">
        <hr></hr>
        <p>Local Storage Things Come Below</p>
        <p>Note : Check console for button click reactions</p>
        <p>TODO : Try showing the values below.</p>
        <p>TODO : Try adding values from an input box</p>
        <button
          type="button"
          className="btn btn-primary"
          onClick={addStuff}
        >
          addStuff
        </button>
        <button
          type="button"
          className="btn btn-primary"
          onClick={showStuff}
        >
          showStuff
        </button>
        <button
            type="button"
            className="btn btn-primary"
            onClick={clearEverything}
          >
            Clear Everything
          </button>        
      </div>
    </div>);
  return stuff;
};

function App() {
  return (
    <div className="App">
      <ShowSomeStuff />
      {/* <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header> */}
    </div>
  );
}

export default App;
