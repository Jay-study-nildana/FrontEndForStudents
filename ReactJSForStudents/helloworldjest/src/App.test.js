import { render, screen } from '@testing-library/react';
import App from './App';
import { create } from 'react-test-renderer'

describe('My first snapshot test',()=>{
  test('testing app button', () => {
  let tree = create(<App />)
  expect(tree.toJSON()).toMatchSnapshot();
})
})
