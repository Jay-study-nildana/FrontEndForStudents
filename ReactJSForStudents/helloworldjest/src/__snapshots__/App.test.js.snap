// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first snapshot test testing app button 1`] = `
<div
  className="App"
>
  <button>
    ShowBro123dfdfdfdfdfdfdfdfdf
  </button>
</div>
`;
